### PR TITLE
Add workflow to run the Mintlify migration

### PR DESCRIPTION
**Pull request by ptgott**
The workflow runs the migration script and pushes to the `mintlify`
branch. It assumes that there is a GitHub app with its ID stored in
`vars` and private key in a secret, and that only this GitHub app has
permissions to push to the `mintlify` branch.

### CODE REVIEW

**Comment by ptgott** [comment]
This PR is based on https://github.com/gravitational/docs/pull/437

File: .github/workflows/mintlify.yml
Comment (wadells) [inline .github/workflows/mintlify.yml:34]: We're currently doing all this on the mintlify branch, as it was checked out immediately above. I think we want to pull in changes from `main` first, then rebuild.

File: migration/base/favicon.png
Comment (wadells) [inline migration/base/favicon.png:1]: Is this intended to be committed? I don't think we want it in `main`.

Comment (ptgott) [reply migration/base/favicon.png:1]: 👍 I'm not able to change my base branch to the fork, so I'll wait until we merge that PR (which should be soon) and rebase this one to `main`.

**Review by wadells — CHANGES_REQUESTED** [review]
Sasha has requested we put this on hold pending discussion around https://mintlify.com/blog/incident-march-13

I won't configure the app/secret needed for minlify to read the docs repo until that discussion is complete.

**Pull request closed without merging**
